app/GraphQL/Mutations/InsertDonation.php

```diff
@@ -21,14 +21,15 @@ class createDeviceToken
      */
     public function resolve($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
     {
-        $user_id = $args['user_id'];
-        $token = $args['token'];
-        $device_model = new Device;
-        $device = $device_model->addOrActivate(array_merge($request->all(), ['user_id' => $user_id, 'token' => $token]));
+       
+        $donation_model = new Donation;
+        $request_dummy = new Request;
+        $donation = $donation_model->add($request_dummy->all());
+        $donation->insertDonation($args['donationData']);
```

Comment:
`$donation->insertDonation($args['donation']);` would be better. Rest of the argument is using the `_` variable naming format - so don't mix formats by doing `donationData`. Here even that is not needed - just use `$args['donation']`